src/js/index.js

```diff
@@ -46,44 +47,58 @@ function handleDifficulty(difficultyChanged) {
 	return false;
 }
 
+function handleSort() {
+	Array.from(contentEl.children)
+		.map(child => contentEl.removeChild(child))
+		.sort(sort[sortingInput.value])
+		.forEach(sortedChild => contentEl.append(sortedChild));
+}
+
 function handleTags() {
 	const tags = selectr.getValue();
 
-	if (tags.length === 0) {
-		history.pushState(null, '', window.location.pathname);
-		return;
-	}
-
 	Array.from(contentEl.querySelectorAll('.item')).forEach(el => {
-		const show = tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), false);
+		const show = (showExpired.checked || !el.classList.contains('tag-expired')) &&
+			tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), tags.length === 0);
 		if (!show) {
 			el.classList.remove('visible');
 		}
 	});
 
-	if (!search) {
-		return;
-	}
-
 	search.set('tags', tags.join(' '));
-	const newRelativePathQuery = `${window.location.pathname}?${search.toString()}`;
-	history.pushState(null, '', newRelativePathQuery);
+
+	search.set('expired', showExpired.checked ? 'true' : '');
 }
 
-function handleSort() {
-	Array.from(contentEl.children)
-		.map(child => contentEl.removeChild(child))
-		.sort(sort[sortingInput.value])
-		.forEach(sortedChild => contentEl.append(sortedChild));
+function updateUrl() {
+	let nextPath = window.location.pathname;
+
+	const emptyParams = [];
+	for (const [key, value] of search) {
+		if (!value.trim()) {
+			emptyParams.push(key);
+		}
+	}
+
+	emptyParams.forEach(param => search.delete(param));
+
+	const queryString = search.toString();
+	if (queryString) {
+		nextPath += `?${queryString}`;
+	}
+
+	history.pushState(null, '', nextPath);
 }
 
 // The cascade is the function which handles calling filtering and sorting swag
 function cascade(force = false) {
 	force |= handleDifficulty(this === filterInput);
-	force |= handleTags(Boolean(this.el));
 	if (force || this === sortingInput) {
```

Comment:
Just noticed this, what was the reasoning for switching the order that `handleTags` and `handleSort` are called?

Reply:
I did it since the logical and visual order in which (I think) we handle everything is:

1. Difficulty
2. Sorting
3. Tags
4. Expired

And so I wanted the code to reflect it.

Reply:
I think there was a reason for putting tags before sorting but I don't remember right now